firefox-ios/Client/Frontend/Browser/SearchEngines/RecentSearchProvider.swift

```diff
@@ -4,66 +4,67 @@
 
 import Common
 import Shared
+import Storage
 
 /// Abstraction for any search client that can return trending searches. Able to mock for testing.
 protocol RecentSearchProvider {
-    var recentSearches: [String] { get }
-    func addRecentSearch(_ term: String)
-    func clearRecentSearches()
+    func addRecentSearch(_ term: String, url: String?)
+    func loadRecentSearches(completion: @escaping ([String]) -> Void)
 }
 
-/// A provider that manages recent search terms for a specific search engine.
+/// A provider that manages recent search terms from a user's history storage.
 struct DefaultRecentSearchProvider: RecentSearchProvider {
```

Comment:
If you are reviewing this file, I would say to review the [Raw](https://github.com/mozilla-mobile/firefox-ios/blob/af920fb3c864968339a13b8b01f13d7875507cb9/firefox-ios/Client/Frontend/Browser/SearchEngines/RecentSearchProvider.swift) file since it might be easier as the whole file is basically rewritten.

Reply:
Thanks @issammani for opening that PR, it'll be helpful for Android too! You 🪨 ! It seems requirements may have change in that we may want to fetch all recent searches (with no limits).  Still TBD, but the method looks great regardless!